lib/reply.js

```diff
@@ -157,16 +157,17 @@ function pumpCallback (reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
-    statusCode = error.status
-  } else if (error.statusCode >= 400) {
-    statusCode = error.statusCode
-  } else {
-    statusCode = statusCode || 500
+
+  if (error) {
```

Comment:
Maybe I'm missing something here, why `if (error)`?

Reply:
Indeed we only have to check it for < 400 case

Reply:
Fixed

Reply:
mmh .. it looks simpler but now it will need 3 checks in most cases...revert?